### PR TITLE
Add a vulkan sample application into gapid.

### PR DESCRIPTION
**Pull request by AWoloszyn**
This will let us have an available application for testing.
This is currently only windows, TODO: expand this to the other platforms in question.

### CODE REVIEW

File: cmd/vulkan_sample/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Copyright (C) 2018 Google Inc.
```
Comment (hevrard) [inline cmd/vulkan_sample/BUILD.bazel]: 2019

File: cmd/vulkan_sample/frag.h [hunk 2]
```diff
@@ -0,0 +1,67 @@
+{0x07230203,0x00010000,0x000d0007,0x00000026,
```
Comment (hevrard) [inline cmd/vulkan_sample/frag.h]: Missing copyright header

Also, I would add a small comment to say this is binary SPIR-V, generated from cmd/vulkan_sample/shader.frag (which I assume is the case)

File: cmd/vulkan_sample/shader.frag [hunk 3]
```diff
@@ -0,0 +1,27 @@
+/* Copyright 2017 Google Inc.
```
Comment (hevrard) [inline cmd/vulkan_sample/shader.frag]: 2019

File: cmd/vulkan_sample/shader.vert [hunk 4]
```diff
@@ -0,0 +1,43 @@
+/* Copyright 2017 Google Inc.
```
Comment (hevrard) [inline cmd/vulkan_sample/shader.vert]: 2019

File: cmd/vulkan_sample/vert.h [hunk 5]
```diff
@@ -0,0 +1,111 @@
+{0x07230203,0x00010000,0x000d0007,0x0000003a,
```
Comment (hevrard) [inline cmd/vulkan_sample/vert.h]: Add header comment: copyright and "SPIR-V generated from shader.vert"

File: cmd/vulkan_sample/main.cpp [hunk 6]
```diff
@@ -0,0 +1,1771 @@
...
+#define REQUIRE_SUCCESS(fn) \
+  do { \
+    if (VK_SUCCESS != fn) { \
+      write_error(kOutHandle, "Vulkan Error: " #fn); \
+      return -1; \
+    } \
+  } while(0)
+
+#define LOAD_INSTANCE_FUNCTION(name, instance) \
+  PFN_##name name = reinterpret_cast<PFN_##name>(vkGetInstanceProcAddr(instance, #name));
```
Comment (hevrard) [inline cmd/vulkan_sample/main.cpp]: Style: put macro definitions outside of main() body ?

**Comment by AWoloszyn** [comment]
Re: GLFW, I don't want to add another dependency into GAPID proper. It's already a mess of dependencies, and for a single .cpp file added, I would avoid more deps.

File: cmd/vulkan_sample/main.cpp [hunk 7]
```diff
@@ -0,0 +1,1771 @@
...
+    break;
+  }
+  if (memory_index ==  properties.memoryTypeCount) {
+    return static_cast<uint32_t>(-1);
+  }
+  return memory_index;
+}
+
+
+int main(int argc, const char** argv) {
```
Comment (yalcinmelihyasin) [inline cmd/vulkan_sample/main.cpp]: I would prefer to split this up to functions but if it is ok for everyone else, I am good too.

**Comment by lpy** [comment]
would prefer to have the main function split into functions like `createInstance`, `createSwapchain`, etc

Assuming Android one is coming in follow-up patch, I am fine with this patch

**Pull request merged**